### PR TITLE
Documented Elyra KFP pipeline steps

### PR DESCRIPTION
**Pull request by oindrillac**
## Related Issues and Dependencies

related to #126 
related to efforts in #124 

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Added documentation on steps to run KFP Pipelines using Elyra which will help team replicate this for other notebooks.

### CODE REVIEW

File: docs/automating-using-elyra.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Running a Kubeflow Workflow using Elyra
+
+## Background
+
+For our data science workflows, we use Jupyter notebooks for data collection, model training and all parts of the machine learning workflow. Some of those tasks are performed sequentially and continuously, requiring us to automate the sequential running of the notebooks using a simple workflow.
+
+Using the [Elyra](https://github.com/elyra-ai/elyra) JupyterLab and the notebook pipelines visual editor, we can create and run our notebooks using Kubeflow Pipelines, which is a platform for building and deploying scalable machine learning (ML) workflows.
+
+An instance of Kubeflow Pipelines is available for use in Operate First.
```
Comment (chauhankaranraj) [inline docs/automating-using-elyra.md:9]:
```suggestion
An [instance of Kubeflow Pipelines](https://www.operate-first.cloud/users/apps/docs/kubeflow/kubeflow-pipelines/README.md) is available for use in Operate First.
```
Would it be worthwhile to add the link to the KFP readme on operate-first site here?

File: docs/automating-using-elyra.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+In this example, we automate a sample Kubeflow workflow using the Elyra UI for the OCP CI Analysis Project.
+
+## Pre-Requisites
+
+1. **Notebook Formatting for Automation**
+
+    * Data Stored in S3 - For notebooks to run in automation and communicate with one another, we store and read the data to and from an S3 bucket.
+    * Automation Flags - The automation flags in the notebook ensure that when the environment variable is configured for AUTOMATION, the notebooks reads and writes from S3.
```
Comment (chauhankaranraj) [inline docs/automating-using-elyra.md:18]:
```suggestion
    * Automation Flags - The automation flags in the notebook ensure that when the environment variable is configured for AUTOMATION, the notebook reads and writes from S3.
```
small typo?

**Review by chauhankaranraj — CHANGES_REQUESTED** [review]

File: docs/automating-using-elyra.md (hunk 2 above)
Comment (oindrillac) [reply docs/automating-using-elyra.md:18]: 👍

**Review by Shreyanand — CHANGES_REQUESTED** [review]
This looks good, thanks for documenting the process in detail. There are some minor comments/suggestions:
1) Could we decrease the size of first screenshot (spawner image)? It appears to be too big in the markdown.
2) Would it make sense to add a line or two about using the yaml of the pipeline and making sure the properties are being reflected there before moving to kubeflow?

**Comment by oindrillac** [comment]
@Shreyanand thanks for the suggestions. Have cropped the image to include the relevant parts. I'm not very keen on applying unique sizing to the images, since as we need to render it on the website, the global config would dictate a uniform sizing anyway.

File: docs/automating-using-elyra.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Running a Kubeflow Workflow using Elyra
+
+## Background
+
+For our data science workflows, we use Jupyter notebooks for data collection, model training and all parts of the machine learning workflow. Some of those tasks are performed sequentially and continuously, requiring us to automate the sequential running of the notebooks using a simple workflow.
+
+Using the [Elyra](https://github.com/elyra-ai/elyra) JupyterLab and the notebook pipelines visual editor, we can create and run our notebooks using Kubeflow Pipelines, which is a platform for building and deploying scalable machine learning (ML) workflows.
+
```
Comment (hemajv) [inline docs/automating-using-elyra.md:8]: would it be good to also include a link to [Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/overview/pipelines-overview/)?

**Review by aakankshaduggal — COMMENTED** [review]
Thanks for putting this together @oindrillac 
/lgtm

File: docs/automating-using-elyra.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+7. Insert inputs for each step/notebook in terms of image runtime, environment variables and resources by going to "Properties".
+
+![properties](assets/images/elyra-3.png)
+
+Any dependencies on other python files and notebooks can be specified here. The files must be present within the same directory.
+The metric notebooks here are dependent on the `metric_template` notebook and hence we specify that as a dependency.
+
+![dependencies](assets/images/elyra-4.png)
+
+We can also check the ``.pipeline` file that was created to check that whether it reflects the parameters we set.
```
Comment (hemajv) [inline docs/automating-using-elyra.md:60]: ``.pipeline`

changed to `.pipeline`?

File: docs/automating-using-elyra.md (hunk 3 above)
Comment (oindrillac) [reply docs/automating-using-elyra.md:8]: @hemajv linked the websites [kubeflow pipelines doc](https://www.operate-first.cloud/users/apps/docs/kubeflow/kubeflow-pipelines/README.md) from which the above is linked.

File: docs/automating-using-elyra.md (hunk 4 above)
Comment (oindrillac) [reply docs/automating-using-elyra.md:60]: ah thanks for the catch

File: docs/automating-using-elyra.md (hunk 3 above)
Comment (hemajv) [reply docs/automating-using-elyra.md:8]: ah okay saw that :+1:

File: docs/automating-using-elyra.md (hunk 4 above)
Comment (oindrillac) [reply docs/automating-using-elyra.md]: fixed 👍

**Comment by hemajv** [comment]
thanks for the great doc! :smiley: 
/lgtm

**Pull request merged**